Engine/SerializerPlugins/ObjectSerializer.cs

```diff
@@ -80,10 +80,18 @@ public virtual bool TryDeserializeObject(XElement element, ITypeData t, Action<o
                     {
                         throw new Exception($"Cannot create instance of {t} and no default value exists.");
                     }
+                  
+                    if (ownerMember.HasAttribute<IFactoryAttribute>())
+                    {
+                        newobj = FactoryAttribute.Create(ownerObj, ownerMember.GetAttribute<IFactoryAttribute>());
+                    }
```

Comment:
It looks like `IFactoryAttribute` would match both a `FactoryAttribute` and `ElementFactoryAttribute`, right? In this example:

```csharp
[ElementFactory("SomeMethod")]
public List<object> MyList { get; set; }
```
We would end up trying to use the element factory to construct MyList, which does not seem correct.

I think we should only check FactoryAttribute here